influx/client.go

```diff
@@ -1,16 +1,198 @@
+// Copyright 2021 InfluxData, Inc. All rights reserved.
+// Use of this source code is governed by MIT
+// license that can be found in the LICENSE file.
+
+// Package influx provides client for InfluxDB server.
 package influx
 
-import "github.com/bonitoo-io/influxdb3-go/influx/configs"
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"mime"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+	"strings"
+)
+
+// Params holds the parameters for creating a new client.
+// The only mandatory field is ServerURL. AuthToken is also important
+// if authentication was not done outside this client.
+type Params struct {
+	// ServerURL holds the URL of the InfluxDB server to connect to.
+	// This must be non-empty. E.g. http://localhost:8086
+	ServerURL string
+
+	// AuthToken holds the authorization token for the API.
+	// This can be obtained through the GUI web browser interface.
+	AuthToken string
+
+	// Organization is name or ID of organization where data (buckets, users, tasks, etc.) belongs to
+	// Optional for InfluxDB Cloud
+	Organization string
 
-// Client provides an interface for interacting with an InfluxDB server, simplifying common operations such as writing, querying.
+	// HTTPClient is used to make API requests.
+	//
+	// This can be used to specify a custom TLS configuration
+	// (TLSClientConfig), a custom request timeout (Timeout),
+	// or other customization as required.
+	//
+	// It HTTPClient is nil, http.DefaultClient will be used.
+	HTTPClient *http.Client
+	// Write Params
+	WriteParams WriteParams
+}
+
+// Client implements an InfluxDB client.
 type Client struct {
-	host     string
-	database string
-	token    string
+	// Configuration params.
+	params Params
```

Comment:
Please rename to configs:
```suggestion
	configs Configs
```